### PR TITLE
fix: collapsible menu slowness

### PR DESCRIPTION
**Pull request by xoRmalka**
Update for #254 #215 

# Description
The collapsible menu on the dashboard page was slow when opening and closing

### CODE REVIEW

**Comment by NoamGaash** [comment]
image

File: src/pages/dashboard/ArrivalByTimeChart/ArrivalByTimeChart.tsx [hunk 1]
```diff
@@ -43,8 +43,35 @@ export default function ArrivalByTimeChart({
     () => data.filter((item) => !operatorId || item.id === operatorId),
     [data, operatorId],
   )
+
+  const [containerWidth, setContainerWidth] = useState<number | undefined>(undefined)
+
+  const chartContainerStyle = {
+    maxWidth: containerWidth ? `${containerWidth - 380}px` : '100%',
```
Comment (NoamGaash) [inline src/pages/dashboard/ArrivalByTimeChart/ArrivalByTimeChart.tsx:50]: https://github.com/hasadna/open-bus-map-search/pull/250#discussion_r1406614005

**Review by NoamGaash — CHANGES_REQUESTED** [review]
lets use responsive containers rather than state maintenance

**Comment by xoRmalka** [comment]
After digging around the web and the documentation 📢 
I found that "ResponsiveContainer" has a "debounce" prop that can be used to control how often the resize event is handled.

@NoamGaash @shootermv

**Review by shootermv — APPROVED** [review]

**Review by NoamGaash — APPROVED** [review]

**Pull request merged**